src-java/nbworker-topology/nbworker-storm-topology/src/main/java/org/openkilda/wfm/topology/nbworker/services/FlowOperationsService.java

```diff
@@ -357,77 +354,140 @@ public Flow updateFlow(FlowOperationsCarrier carrier, FlowPatch flowPatch) throw
+                && !flowPatch.getAllocateProtectedPath().equals(flow.isAllocateProtectedPath());
+
+        updateRequired |= updateRequiredByDiverseFlowIdField(flowPatch, flow);
+
+        updateRequired |= flowPatch.getIgnoreBandwidth() != null
+                && flow.isIgnoreBandwidth() != flowPatch.getIgnoreBandwidth();
+
+        updateRequired |= flowPatch.getEncapsulationType() != null
+                && !flow.getEncapsulationType().equals(flowPatch.getEncapsulationType());
+
+        return UpdateFlowResult.builder()
+                .needUpdateFlow(updateRequired);
+    }
+
+    private boolean updateRequiredByPathComputationStrategy(FlowPatch flowPatch, Flow flow) {
+        boolean changedStrategy = flowPatch.getPathComputationStrategy() != null
                 && !flowPatch.getPathComputationStrategy().equals(flow.getPathComputationStrategy());
         boolean changedMaxLatency = flowPatch.getMaxLatency() != null
                 && !flowPatch.getMaxLatency().equals(flow.getMaxLatency());
         boolean strategyIsMaxLatency =
                 PathComputationStrategy.MAX_LATENCY.equals(flowPatch.getPathComputationStrategy())
-                || flowPatch.getPathComputationStrategy() == null
-                && PathComputationStrategy.MAX_LATENCY.equals(flow.getPathComputationStrategy());
-        updateRequired |= changedMaxLatency && strategyIsMaxLatency;
-
-        // source endpoint
-        updateRequired |= flowPatch.getSourceSwitch() != null
-                && !flow.getSrcSwitch().getSwitchId().equals(flowPatch.getSourceSwitch());
-        updateRequired |= flowPatch.getSourcePort() != null
-                && flow.getSrcPort() != flowPatch.getSourcePort();
-        updateRequired |= flowPatch.getSourceVlan() != null
-                && flow.getSrcVlan() != flowPatch.getSourceVlan();
-
-        // destination endpoint
-        updateRequired |= flowPatch.getDestinationSwitch() != null
-                && !flow.getDestSwitch().getSwitchId().equals(flowPatch.getDestinationSwitch());
-        updateRequired |= flowPatch.getDestinationPort() != null
-                && flow.getDestPort() != flowPatch.getDestinationPort();
-        updateRequired |= flowPatch.getDestinationVlan() != null
-                && flow.getDestVlan() != flowPatch.getDestinationVlan();
+                        || flowPatch.getPathComputationStrategy() == null
+                        && PathComputationStrategy.MAX_LATENCY.equals(flow.getPathComputationStrategy());
+        return changedStrategy || changedMaxLatency && strategyIsMaxLatency;
+    }
 
-        updateRequired |= flowPatch.getBandwidth() != null && flow.getBandwidth() != flowPatch.getBandwidth();
-        updateRequired |= flowPatch.getAllocateProtectedPath() != null
-                && !flowPatch.getAllocateProtectedPath().equals(flow.isAllocateProtectedPath());
+    private boolean updateRequiredBySource(FlowPatch flowPatch, Flow flow) {
```

Comment:
please remove duplicate code, updateRequiredBySource and updateRequiredByDestination are almost the same